File: packages/shared-types/src/dtos/boards/board.dto.ts
import { EntityTimestamps } from '../../types/common.types';
import { TaskDto } from '../tasks/task.dto';

/**
 * Board DTO
 */
export interface BoardDto extends EntityTimestamps {
  id: string;
  name: string;
  slug: string;
  description: string | null;
  color: string;
  projectId: string;
  filePath: string | null;
}

/**
 * Column with tasks for board detail
 */
export interface BoardColumnDto {
  id: string;
  name: string;
  position: number;
  color: string;
  wipLimit: number | null;
  tasks: TaskDto[];
  taskCount: number;
}

/**
 * Board with columns and tasks (GET /boards/:id)
 */
export interface BoardDetailDto extends BoardDto {
  columns: BoardColumnDto[];
  projectName: string;
}

/**
 * Create board request
 */
export interface BoardCreateRequestDto {
  name: string;
  description?: string;
  color?: string;
  projectId: string;
}

/**
 * Update board request
 */
export interface BoardUpdateRequestDto {
  name?: string;
  description?: string;
  color?: string;
}

/**
 * Board list response (GET /boards)
 */
export interface BoardListResponseDto {
  items: BoardDto[];
  total: number;
  page: number;
  limit: number;
}

File: apps/tui/internal/application/dto/board_dto.go
package dto

type BoardDto struct {
	ID          string  `json:"id"`
	Name        string  `json:"name"`
	Slug        string  `json:"slug"`
	Description *string `json:"description"`
	Color       string  `json:"color"`
	ProjectID   string  `json:"projectId"`
	FilePath    *string `json:"filePath"`
	CreatedAt   string  `json:"createdAt"`
	UpdatedAt   string  `json:"updatedAt"`
}

type BoardDetailDto struct {
	BoardDto
	Columns     []BoardColumnDto `json:"columns"`
	ProjectName string           `json:"projectName"`
}

type BoardColumnDto struct {
	ID        string    `json:"id"`
	Name      string    `json:"name"`
	Position  int       `json:"position"`
	Color     string    `json:"color"`
	WipLimit  *int      `json:"wipLimit"`
	Tasks     []TaskDto `json:"tasks"`
	TaskCount int       `json:"taskCount"`
}

type BoardCreateRequest struct {
	Name        string  `json:"name"`
	Description *string `json:"description,omitempty"`
	Color       *string `json:"color,omitempty"`
	ProjectID   string  `json:"projectId"`
}

type BoardUpdateRequest struct {
	Name        *string `json:"name,omitempty"`
	Description *string `json:"description,omitempty"`
	Color       *string `json:"color,omitempty"`
}
